### PR TITLE
Fix Mysql 6.x plugin

### PR DESCRIPTION
**Pull request by candyleer**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
#2797 
___
### Bug fix
- Bug description.

- How to fix?
make it works like 8.x plugin and make it compatible with all 6.x versions and all mode, like replication, loadbalance, failover and single model.
By enhace `ConnectionImpl.getInstance()` method.

and optimize some NPE may occur.

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Fix CI first.

**Comment by wu-sheng** [comment]
Also, run the integration test cases locally, and provide test results here.

**Comment by candyleer** [comment]
Test Report and I update the test case in pull request  https://github.com/SkyAPMTest/agent-auto-integration-testcases/pull/75
# Test Report - 2019-06-01 18:36

- tests  : **53 passed**. **0 failed**
- branch name : **[fix_mysql_6.x_plugin](https://github.com/apache/incubator-skywalking/tree/fix_mysql_6.x_plugin)**
- commit id : **[b91304b9a682d75c8f7dac06886926915cd60abd](https://github.com/apache/incubator-skywalking/commit/b91304b9a682d75c8f7dac06886926915cd60abd)**
- cases branch: **[update_mysql_scenario](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/tree/update_mysql_scenario)**
- cases commit id: **[5e4d4f3e8bd1b651dabbb99b23312aa38e732eb7](https://github.com/SkywalkingTest/skywalking-autotest-scenarios/commit/5e4d4f3e8bd1b651dabbb99b23312aa38e732eb7)**

## Cases List

| Case description | Status | Cases|
|:-----|:-----:|:-----:|
|Mysql| **53 passed. 0 failed**| [click me](#mysql) |

## Mysql

### 
|  Version     | Passed | Failed|
|:------------- |:-------:|:-----:|
| 8.0.15  | :heavy_check_mark:||
| 8.0.14  | :heavy_check_mark:||
| 8.0.13  | :heavy_check_mark:||
| 8.0.12  | :heavy_check_mark:||
| 8.0.11  | :heavy_check_mark:||
| 6.0.6  | :heavy_check_mark:||
| 6.0.5  | :heavy_check_mark:||
| 6.0.4  | :heavy_check_mark:||
| 6.0.3  | :heavy_check_mark:||
| 6.0.2  | :heavy_check_mark:||
| 5.1.45  | :heavy_check_mark:||
| 5.1.44  | :heavy_check_mark:||
| 5.1.43  | :heavy_check_mark:||
| 5.1.42  | :heavy_check_mark:||
| 5.1.41  | :heavy_check_mark:||
| 5.1.40  | :heavy_check_mark:||
| 5.1.39  | :heavy_check_mark:||
| 5.1.38  | :heavy_check_mark:||
| 5.1.37  | :heavy_check_mark:||
| 5.1.36  | :heavy_check_mark:||
| 5.1.35  | :heavy_check_mark:||
| 5.1.34  | :heavy_check_mark:||
| 5.1.33  | :heavy_check_mark:||
| 5.1.32  | :heavy_check_mark:||
| 5.1.31  | :heavy_check_mark:||
| 5.1.30  | :heavy_check_mark:||
| 5.1.29  | :heavy_check_mark:||
| 5.1.28  | :heavy_check_mark:||
| 5.1.27  | :heavy_check_mark:||
| 5.1.26  | :heavy_check_mark:||
| 5.1.25  | :heavy_check_mark:||
| 5.1.24  | :heavy_check_mark:||
| 5.1.23  | :heavy_check_mark:||
| 5.1.22  | :heavy_check_mark:||
| 5.1.21  | :heavy_check_mark:||
| 5.1.20  | :heavy_check_mark:||
| 5.1.19  | :heavy_check_mark:||
| 5.1.18  | :heavy_check_mark:||
| 5.1.17  | :heavy_check_mark:||
| 5.1.16  | :heavy_check_mark:||
| 5.1.15  | :heavy_check_mark:||
| 5.1.14  | :heavy_check_mark:||
| 5.1.13  | :heavy_check_mark:||
| 5.1.12  | :heavy_check_mark:||
| 5.1.11  | :heavy_check_mark:||
| 5.1.10  | :heavy_check_mark:||
| 5.1.9  | :heavy_check_mark:||
| 5.1.8  | :heavy_check_mark:||
| 5.1.6  | :heavy_check_mark:||
| 5.1.5  | :heavy_check_mark:||
| 5.1.4  | :heavy_check_mark:||
| 5.1.3  | :heavy_check_mark:||
| 5.1.2  | :heavy_check_mark:||

File: apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCDriverInterceptor.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
     @Override public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
         Class<?>[] argumentsTypes, Object ret) throws Throwable {
-        if (ret != null) {
+        if (ret instanceof EnhancedInstance) {
```
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCDriverInterceptor.java]:
```suggestion
        if (ret != null && ret instanceof EnhancedInstance) {
```

File: apm-sniffer/apm-sdk-plugin/mysql-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v6/ConnectionCreateNewInterceptor.java [hunk 2]
```diff
@@ -6,49 +6,51 @@
...
     @Override
-    public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes,
-        Object ret) {
-        ConnectionUrl connectionUrl = (ConnectionUrl)allArguments[0];
-
-        StringBuilder hosts = new StringBuilder();
-        for (HostInfo info : connectionUrl.getHostsList()) {
-            hosts.append(info.getHost()).append(":").append(info.getPort()).append(",");
+    public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes, Object ret) {
+        if (ret instanceof EnhancedInstance) {
```
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/mysql-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v6/ConnectionCreateNewInterceptor.java]:
```suggestion
        if (ret !=null && ret instanceof EnhancedInstance) {
```

File: apm-sniffer/apm-sdk-plugin/mysql-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v6/ConnectionCreateOldInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+    @Override
+    public void beforeMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes, MethodInterceptResult result) {
+
+    }
+
+    @Override
+    public Object afterMethod(Class clazz, Method method, Object[] allArguments, Class<?>[] parameterTypes, Object ret) {
+        if (ret instanceof EnhancedInstance) {
```
Comment (ascrutae) [inline apm-sniffer/apm-sdk-plugin/mysql-6.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/v6/ConnectionCreateOldInterceptor.java]:
```suggestion
        if (ret != null && ret instanceof EnhancedInstance) {
```

**Pull request merged**